tesseract_process_managers/include/tesseract_process_managers/core/process_planning_server.h

```diff
@@ -163,12 +197,36 @@ class ProcessPlanningServer
    */
   ProfileDictionary::ConstPtr getProfiles() const;
 
+  /**
+   * @brief Queries the number of worker threads (can be zero)
+   * @param name The name of the executor to get worker count for
+   */
+  long getWorkerCount(const std::string& name = PRIMARY_EXECUTOR_NAME) const;
+
+  /**
+   * @brief Queries the number of running tasks at the time of this call
+   * When a taskflow is submitted to an executor, a topology is created to store
+   * runtime metadata of the running taskflow.
+   * @param name The name of the executor to get task count for
+   */
+  long getTaskCount(const std::string& name = PRIMARY_EXECUTOR_NAME) const;
+
 protected:
+  mutable std::shared_mutex mutex_;
```

Comment:
Maybe a more descriptive name? It looks like this mutex is specifically for the executor map

Reply:
It is primarily used for all three maps withing the class. 
``` c++
/** @brief A map of executors which are thread safe */
  std::unordered_map<std::string, std::shared_ptr<tf::Executor>> executors_;

  /** @brief A map of profile observers */
  std::unordered_map<std::string, std::shared_ptr<tf::TFProfObserver>> profile_observers_;

  /** @brief A map of process planners */
  std::unordered_map<std::string, TaskflowGenerator::UPtr> process_planners_;
```

Reply:
Ok. Well this obviously doesn't really matter, so I think it's fine.